lightningd/plugin_control.c

```diff
@@ -71,143 +52,27 @@ struct command_result *plugin_cmd_all_complete(struct plugins *plugins,
-                                             const jsmntok_t *idtok,
-                                             struct dynamic_plugin *dp)
-{
-	if (!plugin_parse_getmanifest_response(buffer, toks, idtok, dp->plugin))
-		return was_pending(plugin_dynamic_error(dp, "Gave a bad response to getmanifest"));
-
-	if (!dp->plugin->dynamic)
-		return was_pending(plugin_dynamic_error(dp, "Not a dynamic plugin"));
-
-	/* We got the manifest, now send the init message */
-	dp->plugin->plugin_state = NEEDS_INIT;
-	plugin_dynamic_config(dp);
-}
-
-/**
- * This starts a plugin : spawns the process, connect its stdout and stdin,
- * then sends it a getmanifest request.
- */
-static struct command_result *plugin_start(struct dynamic_plugin *dp)
-{
-	int stdin, stdout;
-	mode_t prev_mask;
-	char **p_cmd;
-	struct jsonrpc_request *req;
-	struct plugin *p = dp->plugin;
-
-	p->dynamic = false;
-	p_cmd = tal_arrz(NULL, char *, 2);
-	p_cmd[0] = p->cmd;
-	/* In case the plugin create files, this is a better default. */
-	prev_mask = umask(dp->cmd->ld->initial_umask);
-	p->pid = pipecmdarr(&stdin, &stdout, &pipecmd_preserve, p_cmd);
-	umask(prev_mask);
-	if (p->pid == -1)
-		return plugin_dynamic_error(dp, "Error running command");
-	else
-		log_debug(dp->cmd->ld->plugins->log, "started(%u) %s", p->pid, p->cmd);
-	tal_free(p_cmd);
-	p->buffer = tal_arr(p, char, 64);
-	p->stop = false;
-	/* Give the plugin 20 seconds to respond to `getmanifest`, so we don't hang
-	 * too long on the RPC caller. */
-	p->timeout_timer = new_reltimer(dp->cmd->ld->timers, dp,
-	                                time_from_sec((20)),
-	                                plugin_dynamic_timeout, dp);
-
-	/* Besides the timeout we could also have the plugin crash before
-	 * completing the handshake. In that case we'll get notified and we
-	 * can clean up the `struct dynamic_plugin` and return an appropriate
-	 * error.
-	 *
-	 * The destructor is deregistered in the following places:
-	 *
-	 *  - plugin_dynamic_error in case of a timeout or a crash
-	 *  - plugin_dynamic_config_callback if the handshake completes
-	 */
-	tal_add_destructor2(p, plugin_dynamic_crash, dp);
-
-	/* Create two connections, one read-only on top of the plugin's stdin, and one
-	 * write-only on its stdout. */
-	io_new_conn(p, stdout, plugin_stdout_conn_init, p);
-	io_new_conn(p, stdin, plugin_stdin_conn_init, p);
-	req = jsonrpc_request_start(p, "getmanifest", p->log,
-	                            plugin_dynamic_manifest_callback, dp);
-	jsonrpc_request_end(req);
-	plugin_request_send(p, req);
-	p->plugin_state = AWAITING_GETMANIFEST_RESPONSE;
-	return command_still_pending(dp->cmd);
-}
-
 /**
  * Called when trying to start a plugin through RPC, it starts the plugin and
  * will give a result 20 seconds later at the most.
  */
 static struct command_result *
 plugin_dynamic_start(struct command *cmd, const char *plugin_path)
 {
-	struct dynamic_plugin *dp;
```

Comment:
The comment still references 20 seconds.

Reply:
Couldn't add the review comment on that line since it'd confuse GH to have review comments on unchanged lines.